### PR TITLE
Fixes #16734: Update agent to CFEngine 3.15.0

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/16734

### CODE REVIEW

File: rudder-agent/SOURCES/patches/cfengine/13310-disable-openssl-engine.patch [hunk 1]
```diff
@@ -1,13 +0,0 @@
-diff --git a/libutils/libcrypto-compat.c b/libutils/libcrypto-compat.c
-index bf39f574f..5d29b7cfb 100644
---- a/libutils/libcrypto-compat.c
-+++ b/libutils/libcrypto-compat.c
-@@ -14,7 +14,7 @@
- #if OPENSSL_VERSION_NUMBER < 0x10100000L
- 
- #include <string.h>
--#include <openssl/engine.h>
-+//#include <openssl/engine.h>
```
Comment (peckpeck) [inline rudder-agent/SOURCES/patches/cfengine/13310-disable-openssl-engine.patch]: Has this been removed by cfengine ?

Comment (peckpeck) [reply rudder-agent/SOURCES/patches/cfengine/13310-disable-openssl-engine.patch]: In fact thfile still exists, it has just changed its location into libntech/libutils/libcrypto-compat.c

Comment (amousset) [reply rudder-agent/SOURCES/patches/cfengine/13310-disable-openssl-engine.patch]: engine.h is not included anymore

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Rebased for 3.15.1

File: rudder-agent/SOURCES/patches/cfengine/13310-disable-openssl-engine.patch (hunk 1 above)
Comment (amousset) [reply rudder-agent/SOURCES/patches/cfengine/13310-disable-openssl-engine.patch]: still there indeed.

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
OK, squash merging this PR

**Pull request merged**